bin/propolis-server/src/lib/initializer.rs

```diff
@@ -982,100 +942,48 @@ impl<'a> MachineInitializer<'a> {
-            proc_type: type4::ProcType::Central,
-            proc_family,
-            proc_manufacturer,
-            proc_id,
-            proc_version: proc_version.try_into().unwrap_or_default(),
-            status: type4::ProcStatus::Enabled,
-            // unknown
-            proc_upgrade: 0x2,
-            // make core and thread counts equal for now
-            core_count: self.properties.vcpus,
-            core_enabled: self.properties.vcpus,
-            thread_count: self.properties.vcpus,
-            proc_characteristics: type4::Characteristics::IS_64_BIT
-                | type4::Characteristics::MULTI_CORE,
-            ..Default::default()
-        };
-
-        let memsize_bytes = (self.properties.memory as usize) * MB;
-        let mut smb_type16 = smbios::table::Type16 {
-            location: type16::Location::SystemBoard,
-            array_use: type16::ArrayUse::System,
-            error_correction: type16::ErrorCorrection::Unknown,
-            num_mem_devices: 1,
-            ..Default::default()
-        };
-        smb_type16.set_max_capacity(memsize_bytes);
-        let phys_mem_array_handle = 0x1600.into();
-
-        let mut smb_type17 = smbios::table::Type17 {
-            phys_mem_array_handle,
-            // Unknown
-            form_factor: 0x2,
-            // Unknown
-            memory_type: 0x2,
-            ..Default::default()
-        };
-        smb_type17.set_size(Some(memsize_bytes));
-
-        let smb_type32 = smbios::table::Type32::default();
-
         // With "only" types 0, 1, 4, 16, 17, and 32, we are technically missing
         // some (types 3, 7, 9, 19) of the data required by the 2.7 spec.  The
         // data provided here were what we determined was a reasonable
         // collection to start with.  Should further requirements arise, we may
         // expand on it.
         let mut smb_tables = smbios::Tables::new(0x7f00.into());
-        smb_tables.add(0x0000.into(), &smb_type0).unwrap();
-        smb_tables.add(0x0100.into(), &smb_type1).unwrap();
-        smb_tables.add(0x0300.into(), &smb_type4).unwrap();
-        smb_tables.add(phys_mem_array_handle, &smb_type16).unwrap();
-        smb_tables.add(0x1700.into(), &smb_type17).unwrap();
-        smb_tables.add(0x3200.into(), &smb_type32).unwrap();
+        smb_tables.add(0x0000.into(), &smbios_params.table_type0()).unwrap();
+        smb_tables.add(0x0100.into(), &smbios_params.table_type1()).unwrap();
+        smb_tables.add(0x0300.into(), &smbios_params.table_type4()).unwrap();
+        let phys_mem_array_handle = 0x1600.into();
+        smb_tables
+            .add(phys_mem_array_handle, &smbios_params.table_type16())
+            .unwrap();
+        smb_tables
+            .add(
+                0x1700.into(),
+                &smbios_params.table_type17(phys_mem_array_handle),
+            )
+            .unwrap();
+        smb_tables.add(0x3200.into(), &smbios_params.table_type32()).unwrap();
```

Comment:
@pfmooney how's this seem? my thinking is that the `SmbiosParams::table_*` are "this is probably a reasonable way to put the tables together", and if we find value in changing any of the defaults we'd hoist them into fields of `SmbiosParameters`.